lib/validations.js

```diff
@@ -760,10 +760,21 @@ function ValidationError(obj) {
 util.inherits(ValidationError, Error);
 
 var errorHasStackProperty = !!(new Error).stack;
+var inspect = require('util').inspect;
+
+function formatProperty(obj, propertyName) {
+  var value = obj[propertyName];
+  if ((typeof value) === 'function')
+    return propertyName
+  else if ((typeof value) === 'string')
+    return propertyName + ': ' + inspect(value.substring(0, 32));
+  else
+    return propertyName + ': ' + inspect(value);
+}
 
```

Comment:
Handling non-scalar values (objects, arrays, functions) will be always tricky. You can take a look at [chai.js](https://github.com/chaijs/chai/blob/5fc486b5733d67a02ead53de6557741d4c57c276/lib/chai/utils/objDisplay.js#L31-L42) which basically prints only top-level keys of the object.

As for function values, `JSON.stringify` removes them from the output, which solves the problem. On the other hand, it does not handle circular references, which may be a problem.